src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -691,11 +691,6 @@ internal void EnsureStoreIsValid()
             if (Disposed)
                 throw new ObjectDisposedException(null, SR.IsolatedStorage_StoreNotOpen);
 
-            if (IsDeleted)
-            {
-                throw new IsolatedStorageException(SR.IsolatedStorage_StoreNotOpen);
-            }
-
             if (_closed)
```

Comment:
Is it always true that when _closed is True, IsDeleted is also true? i.e. are we sure that the IsDeleted check isn't catching any cases that aren't caught by checking _closed?